src/openrct2-ui/drawing/engines/opengl/OpenGLDrawingEngine.cpp

```diff
@@ -676,7 +676,14 @@ void OpenGLDrawingContext::DrawLine(DrawPixelInfo& dpi, uint32_t colour, const S
     command.depth = _drawCount++;
 }
 
-void OpenGLDrawingContext::DrawSprite(DrawPixelInfo& dpi, const ImageId imageId, int32_t x, int32_t y)
+template<typename type>
+type EuclideanRemainder(const type a, const type b)
+{
+    const type r = a % b;
+    return r >= 0 ? r : r + b;
+};
```

Comment:
We ordinarily annotate template types as T but I think we can also simplify this by just using auto
```suggestion
static auto EuclideanRemainder(const auto a, const auto b)
{
    const auto r = a % b;
    return r >= 0 ? r : r + b;
};
```